coremltools/converters/mil/mil/ops/defs/iOS15/elementwise_unary.py

```diff
@@ -890,7 +890,7 @@ def get_cast_value(cls, input_var, dtype_val):
                 return np.array(result)
             return None
 
-        if not types.is_tensor(input_var.sym_type):
-            return input_var.val.astype(dtype=string_to_nptype(dtype_val))
-        else:
+        if isinstance(input_var.val, float) or types.is_tensor(input_var.sym_type):
```

Comment:
I would be nice to have unit tests for this change, i.e. a test which fails without this change but passes with this change.

Reply:
Done in https://github.com/apple/coremltools/pull/2087/commits/1137b73599ec3e8589f6ae465a7390fab59600c3.
See my comment in the pull request for more details.